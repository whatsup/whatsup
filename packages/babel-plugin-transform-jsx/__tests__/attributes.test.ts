import { transform } from './__config__/transform'

describe('transform attributes', function () {
    it('Should transform key prop', function () {
        expect(transform('<div key="Key" />')).toMatchSnapshot()
    })

    it('Should transform simple props', function () {
        expect(transform('<div prop1="value1" />')).toMatchSnapshot()
    })

    it('Should transform boolean prop', function () {
        expect(transform('<div focused />')).toMatchSnapshot()
    })

    it('Should transform props with expressions', function () {
        expect(transform('<div prop1={1 + 1} />')).toMatchSnapshot()
    })

    it('Should transform props spread', function () {
        expect(transform('<div {...props} />')).toMatchSnapshot()
    })

    it('Should transform different props variations', function () {
        expect(transform('<div key="Key" prop1="value1" prop2={2 + 2} {...props} />')).toMatchSnapshot()
    })

    it('Should transform different props with ref', function () {
        expect(transform('<div ref={myRef} key="Key" prop1="value1" prop2={2 + 2} {...props} />')).toMatchSnapshot()
    })

    it('Should transform different props with ref onMount onUnmount', function () {
        expect(
            transform(
                '<div ref={myRef} onMount={onMount} onUnmount={onUnmount} key="Key" prop1="value1" prop2={2 + 2} {...props} />'
            )
        ).toMatchSnapshot()
    })

    it('Should transform namespaced name', function () {
        expect(transform('<div css:blue />')).toMatchSnapshot()
    })

    it('Should transform cssx var', function () {
        expect(transform('<div css:$var />')).toMatchSnapshot()
    })
})
