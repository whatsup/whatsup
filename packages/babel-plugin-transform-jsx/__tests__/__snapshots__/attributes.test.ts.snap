// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform attributes Should transform boolean prop 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  focused: true
}];"
`;

exports[`transform attributes Should transform cssx var 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  \\"css:$var\\": true
}];"
`;

exports[`transform attributes Should transform different props variations 1`] = `
"[\\"<0~UniqueId\\" + \\"~\\" + \\"Key\\" + \\">\\", \\"div\\", {
  prop1: \\"value1\\",
  prop2: 2 + 2,
  ...props
}];"
`;

exports[`transform attributes Should transform different props with ref 1`] = `
"[\\"<0~UniqueId\\" + \\"~\\" + \\"Key\\" + \\">\\", \\"div\\", {
  ref: myRef,
  prop1: \\"value1\\",
  prop2: 2 + 2,
  ...props
}];"
`;

exports[`transform attributes Should transform different props with ref onMount onUnmount 1`] = `
"[\\"<0~UniqueId\\" + \\"~\\" + \\"Key\\" + \\">\\", \\"div\\", {
  ref: myRef,
  onMount: onMount,
  onUnmount: onUnmount,
  prop1: \\"value1\\",
  prop2: 2 + 2,
  ...props
}];"
`;

exports[`transform attributes Should transform key prop 1`] = `"[\\"<0~UniqueId\\" + \\"~\\" + \\"Key\\" + \\">\\", \\"div\\", {}];"`;

exports[`transform attributes Should transform namespaced name 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  \\"css:blue\\": true
}];"
`;

exports[`transform attributes Should transform props spread 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", { ...props
}];"
`;

exports[`transform attributes Should transform props with expressions 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  prop1: 1 + 1
}];"
`;

exports[`transform attributes Should transform simple props 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  prop1: \\"value1\\"
}];"
`;
